src/core/Akka.Persistence/Journal/AsyncWriteProxy.cs

```diff
@@ -384,20 +393,24 @@ public override Task ReplayMessagesAsync(IActorContext context, string persisten
         /// </summary>
         /// <param name="persistenceId">TBD</param>
         /// <param name="fromSequenceNr">TBD</param>
+        /// <param name="cancellationToken"><see cref="CancellationToken"/> used to signal cancelled snapshot operation</param>
         /// <exception cref="TimeoutException">
         /// This exception is thrown when the store has not been initialized.
         /// </exception>
         /// <returns>TBD</returns>
-        public override Task<long> ReadHighestSequenceNrAsync(string persistenceId, long fromSequenceNr)
+        public override Task<long> ReadHighestSequenceNrAsync(
+            string persistenceId,
+            long fromSequenceNr,
+            CancellationToken cancellationToken)
         {
             if (_store == null)
-                return StoreNotInitialized<long>();
+                return StoreNotInitialized<long>(); 
 
-            return _store.Ask<AsyncWriteTarget.ReplaySuccess>(new AsyncWriteTarget.ReplayMessages(persistenceId, 0, 0, 0), Timeout)
+            return _store.Ask<AsyncWriteTarget.ReplaySuccess>(new AsyncWriteTarget.ReplayMessages(persistenceId, 0, 0, 0), Timeout, cancellationToken)
```

Comment:
Example of how an internal `Ask()` operation is being guarded now, It can both be cancelled by the defined `Timeout` parameter and cancellation token from the `CircuitBreaker`.